lib/attrsets.nix

```diff
@@ -1208,6 +1208,102 @@ rec {
     in
     recurse [ ] set;
 
+  /**
+    Recursively apply a function to each leaf attribute of the potentially
+    nested attribute set and combine the results in a list.
+
+    The first argument to the mapping function is a list of the attribute names
+    that form the path to the leaf attribute. The second argument will never
+    be an attribute set.
+
+    # Inputs
+
+    `f`
+
+    : A function, given an attribute’s path and value, returns a new value.
+
+    `attrs`
+
+    : Attribute set to map over.
+
+    # Type
+
+    ```
+    mapAttrsToListRecursive :: ([String] -> a -> b) -> AttrSet -> [b]
+    ```
+
+    # Examples
+    :::{.example}
+    ## `lib.attrsets.mapAttrsToListRecursive` usage example
+
+    ```nix
+    mapAttrsToListRecursive (path: value: "${concatStringsSep "." path}=${value}")
+      { n = { a = "A"; m = { b = "B"; c = "C"; }; }; d = "D"; }
+    => [ "n.a=A" "n.m.b=B" "n.m.c=C" "d=D" ]
+    ```
+    :::
+  */
+  mapAttrsToListRecursive =
+    f:
+    set:
+    mapAttrsToListRecursiveCond (path: as: true) f set;
+
+  /**
+    Like `mapAttrsToListRecursive`, but it takes an additional predicate to
+    decide whether to recurse into an attribute set. If the predicate returns
+    false, `mapAttrsToListRecursiveCond` does not recurse, but instead applies
+    the mapping function. If the predicate returns true, it does recurse
+    without applying the mapping function.
+
+    The first argument to the predicate is a list of the attribute names
+    that form the path to the leaf attribute. The second argument is the
```

Comment:
Leaf attribute has not been defined. It seems like a very useful concept, and I've also used it in my comments and suggestions, but the reader is unaware.

I think you could clarify the documentation a lot by making a high level description like:
Determine the nodes of a tree built out of attribute sets, by applying `pred`, then apply a function to the leaves, and return the results of `f` into a list, ordered lexicographically by their attribute path.

Reply:
Thank you for the input. This was really useful.